conjure-go-client/httpclient/internal/refreshingclient/transport.go

```diff
@@ -40,11 +40,33 @@ type TransportParams struct {
 	HTTP2PingTimeout      time.Duration
 }
 
-func NewRefreshableTransport(ctx context.Context, p RefreshableTransportParams, tlsConfig *tls.Config, dialer ContextDialer) http.RoundTripper {
+func NewRefreshableTransport(ctx context.Context, p RefreshableTransportParams, tlsConfig refreshable.Refreshable, dialer ContextDialer) http.RoundTripper {
+	curTLSConfig := func() *tls.Config {
+		if tlsConfig == nil {
+			return nil
+		}
+		return tlsConfig.Current().(*tls.Config)
+	}
+
+	transport := refreshable.NewDefaultRefreshable(newTransport(ctx, p.CurrentTransportParams(), curTLSConfig(), dialer))
+
+	p.SubscribeToTransportParams(func(params TransportParams) {
+		if err := transport.Update(newTransport(ctx, params, curTLSConfig(), dialer)); err != nil {
+			svc1log.FromContext(ctx).Error("Failed to update HTTP transport with transport params", svc1log.Stacktrace(err))
+		}
+	})
+
+	if tlsConfig != nil {
+		tlsConfig.Subscribe(func(tlsConfI interface{}) {
```

Comment:
The general concern I had with the original implementation was this subscription because `*tls.Config` is not compatible with `reflect.DeepEqual` (it contains closures in struct fields) so other changes could cause us to rebuild our transport (and lose shared connections)